rskj-core/src/main/java/org/ethereum/core/BlockHeader.java

```diff
@@ -303,6 +303,11 @@ public Keccak256 getHash() {
         return this.getEncoded(true, true);
     }
 
+    public byte[] getEncodedForBlockHash() {
```

Comment:
I think it's better to rename the current `getEncoded` as something like `getFullEncoded`, and then make this method to be the proper `getEncoded` one. This change should be done as a separate commit at the beginning of this PR